inference-engine/src/plugin_api/cpp_interfaces/impl/ie_infer_async_request_thread_safe_default.hpp

```diff
@@ -72,12 +73,21 @@ class AsyncInferRequestThreadSafeDefault : public AsyncInferRequestThreadSafeInt
      */
     AsyncInferRequestThreadSafeDefault(const InferRequestInternal::Ptr& request,
                                        const ITaskExecutor::Ptr& taskExecutor,
-                                       const ITaskExecutor::Ptr& callbackExecutor)
-        : _syncRequest {request},
-          _requestExecutor {taskExecutor},
-          _callbackExecutor {callbackExecutor},
-          _pipeline {{taskExecutor, [this] {_syncRequest->Infer();}}},
-          _syncPipeline{{std::make_shared<ImmediateExecutor>(), [this] {_syncRequest->Infer();}}} {
+                                       const ITaskExecutor::Ptr& callbackExecutor) :
+        _syncRequest {request},
+        _requestExecutor {taskExecutor},
+        _callbackExecutor {callbackExecutor},
+        _pipeline {{taskExecutor, [this] {_syncRequest->Infer();}}},
+        _syncPipeline{{std::make_shared<ImmediateExecutor>(), [this] {_syncRequest->Infer();}}} {
+        auto streamsExecutor = std::dynamic_pointer_cast<IStreamsExecutor>(taskExecutor);
+        if (streamsExecutor != nullptr) {
+            struct ImmediateStreamsExecutor : public InferenceEngine::ITaskExecutor {
+                explicit ImmediateStreamsExecutor(const IStreamsExecutor::Ptr& streamsExecutor) : _streamsExecutor{streamsExecutor} {}
+                void run(InferenceEngine::Task task) override {_streamsExecutor->Execute(std::move(task));}
+                IStreamsExecutor::Ptr _streamsExecutor;
+            };
+            _syncPipeline = {{std::make_shared<ImmediateStreamsExecutor>(std::move(streamsExecutor)), [this] {_syncRequest->Infer();}}};
```

Comment:
how changing the executor type fixes this issue?  Can we add beh tests for this use case?

Reply:
It dose not fix the issue. 
This change makes synchronous inference execution context use same properties as for asynchronous inference. It is needed for CPU plugins if we want stop using SyncUsingAsync function and use just default Synchronous inference implementation.

Reply:
What beh test you suppose?